src/common/multi_file_reader.cpp

```diff
@@ -387,4 +452,108 @@ void UnionByName::CombineUnionTypes(const vector<string> &col_names, const vecto
 	}
 }
 
+bool MultiFileReaderOptions::AutoDetectHivePartitioningInternal(const vector<string> &files) {
+	std::unordered_set<string> uset;
+	idx_t splits_size;
+	{
+		//	front file
+		auto splits = StringUtil::Split(files.front(), FileSystem::PathSeparator());
+		splits_size = splits.size();
+		if (splits.size() < 2) {
+			return false;
+		}
+		for (auto it = splits.begin(); it != std::prev(splits.end()); it++) {
+			auto part = StringUtil::Split(*it, "=");
+			if (part.size() == 2) {
+				uset.insert(part.front());
+			}
+		}
+	}
+	if (uset.empty()) {
+		return false;
+	}
+	for (auto &file : files) {
+		auto splits = StringUtil::Split(file, FileSystem::PathSeparator());
+		if (splits.size() != splits_size) {
+			return false;
+		}
+		for (auto it = splits.begin(); it != std::prev(splits.end()); it++) {
+			auto part = StringUtil::Split(*it, "=");
+			if (part.size() == 2) {
```

Comment:
nitpick: nested if statements should either be written using return early pattern mentioned in other comment or using `if (<cond_a> && <cond_b>)`